src/services/completions.ts

```diff
@@ -1904,6 +1904,40 @@ namespace ts.Completions {
             return GlobalsSearch.Success;
         }
 
+        /**
+         * Adds local declarations for completions in named exports:
+         *
+         *   export { | };
+         *
+         * Does not check for the absence of a module specifier (`export {} from "./other"`)
+         * because `tryGetImportOrExportClauseCompletionSymbols` runs first and handles that,
+         * preventing this function from running.
+         */
+        function tryGetLocalNamedExportCompletionSymbols(): GlobalsSearch {
+            const namedExports = contextToken.kind === SyntaxKind.OpenBraceToken || contextToken.kind === SyntaxKind.CommaToken
+                ? tryCast(contextToken.parent, isNamedExports)
+                : undefined;
+
+            if (!namedExports) {
+                return GlobalsSearch.Continue;
+            }
+
+            const localsBlock = namedExports.parent.parent;
+            if (localsBlock.kind !== SyntaxKind.SourceFile && localsBlock.kind !== SyntaxKind.ModuleBlock) {
+                return GlobalsSearch.Fail;

```

Comment:
I'm not sure I understand the difference between `Fail` and `Continue`, but I would naively have expected this to be `Continue`.

Reply:
Returning `Continue` would allow the list to be populated with globals, while `Fail` makes the list empty. This condition represents a grammar error (export declarations must be at the top level), so it sort of doesn’t matter, but globals (and non-top-level locals) are definitely wrong to include in the list. Arguably it could be better to just `findAncestor` the module block or source file and offer the same completions you’d get if the export declaration were in a valid location... I guess I’ll do that; it will remove a couple lines of code.